das/daser.go

```diff
@@ -143,6 +157,14 @@ func (d *DASer) sample(ctx context.Context, sub header.Subscription, checkpoint
 			if err == context.Canceled {
 				return
 			}
+			var byzantineErr *ipld.ErrByzantine
+			if errors.As(err, &byzantineErr) {
```

Comment:
I'm wondering if we shouldn't fix https://github.com/celestiaorg/celestia-node/issues/787 before proceeding with merging this PR. Because it's likely we will see these fake byzantineErrs popping up in the network and being gossiped to each other as well. 
At the very least we should hold off with cutting a release that contains fraud service without fixing #787 first.